### DISCUSSION TITLE
With a bunch of attributes from the `[Match]` section reused in other specific sections, what is actually the use of this section?

### DISCUSSION THREAD

**Discussion by gridhead** [systemd/mkosi#2311] [Q&A] [answered]
While the other sections are named appropriately and their name, to a great extent, describes their purpose - the one section that I am confused about is the `[Match]` section, especially in the light of a bunch of attributes there reused in other sections.

Could you please help me understand the purpose of the `[Match]` section?

Thanks in advance!

**Comment by daandemeyer** [comment]
The "Configuration" section in the documentation should explain this in detail.

**Comment by gridhead** [comment]
Thanks, @DaanDeMeyer.

The documentation is indeed elaborate but I think it is me who is still struggling to understand the application of such an option. Please confirm if I understand it correctly - Is it used for having multiple `profiles` of the same image configuration so as to ensure that there's less repetition and more pluggability in the configuration files? (Or did I understand it wrong? ;P)

**Reply by daandemeyer** [reply, marked as answer]
Take a look at the default image configuration in the mkosi repo itself. We make liberal use of Match for various purposes.
